ios/Fabric/RNCPagerViewComponentView.mm

```diff
@@ -0,0 +1,366 @@
+#ifdef RCT_NEW_ARCH_ENABLED
+
+#import <Foundation/Foundation.h>
+#import "RNCPagerViewComponentView.h"
+#import <react/renderer/components/RNCViewPager/ComponentDescriptors.h>
+#import <react/renderer/components/RNCViewPager/EventEmitters.h>
+#import <react/renderer/components/RNCViewPager/Props.h>
+#import <react/renderer/components/RNCViewPager/RCTComponentViewHelpers.h>
+
+#import "RCTFabricComponentsPlugins.h"
+#import "React/RCTConversions.h"
+
+using namespace facebook::react;
+
+@interface RNCPagerViewComponentView () <RCTRNCViewPagerViewProtocol, UIPageViewControllerDataSource, UIPageViewControllerDelegate, UIScrollViewDelegate>
+@end
+
+@implementation RNCPagerViewComponentView {
+    UIScrollView *scrollView;
+}
+
+- (void)initializeNativePageViewController {
+    const auto &viewProps = *std::static_pointer_cast<const RNCViewPagerProps>(_props);
+    NSDictionary *options = @{ UIPageViewControllerOptionInterPageSpacingKey: @(viewProps.pageMargin) };
+    UIPageViewControllerNavigationOrientation orientation = UIPageViewControllerNavigationOrientationHorizontal;
+    switch (viewProps.orientation) {
+        case RNCViewPagerOrientation::Horizontal:
+            orientation = UIPageViewControllerNavigationOrientationHorizontal;
+            break;
+        case RNCViewPagerOrientation::Vertical:
+            orientation = UIPageViewControllerNavigationOrientationVertical;
+            break;
+    }
+    _nativePageViewController = [[UIPageViewController alloc]
+                                 initWithTransitionStyle: UIPageViewControllerTransitionStyleScroll
+                                 navigationOrientation:orientation
+                                 options:options];
+    _nativePageViewController.dataSource = self;
+    _nativePageViewController.delegate = self;
+    _nativePageViewController.view.frame = self.frame;
+    [self addSubview:_nativePageViewController.view];
+    
+    for (UIView *subview in _nativePageViewController.view.subviews) {
+        if([subview isKindOfClass:UIScrollView.class]){
+            ((UIScrollView *)subview).delegate = self;
+            ((UIScrollView *)subview).delaysContentTouches = NO;
+            scrollView = (UIScrollView *)subview;
+        }
+    }
+}
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    if (self = [super initWithFrame:frame]) {
+        static const auto defaultProps = std::make_shared<const RNCViewPagerProps>();
+        _props = defaultProps;
+        _nativeChildrenViewControllers = [[NSMutableArray alloc] init];
+        _currentIndex = -1;
+        _layoutDirection = @"ltr";
+        _overdrag = NO;
+        
+        [self initializeNativePageViewController];
+    }
+    
+    return self;
+}
+
+
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+    //Workaround to fix incorrect frame issue
+    [self goTo:_currentIndex animated:NO];
+}
+
+-(void)prepareForRecycle{
+    [super prepareForRecycle];
```

Comment:
Do you know, when this method is called ?

Reply:
https://github.com/facebook/react-native/blob/f3db6cc52792e3006a16408df4ae40f3aee19a86/React/Fabric/Mounting/RCTComponentViewProtocol.h#L112